test/e2e/edu-1990n-helpers.js

```diff
@@ -0,0 +1,16 @@
+// I really don't want to duplicate this, but the lack of view:veteranId really
```

Comment:
I was trying to refactor the shared one first, but after much difficulty, I broke down and just copied it out. I tried making an `exists()` function for the client to fill in information if an element exists. I tried filling in the `view:veteranId` version first and if that fails, the field sans `view:veteranId`, but that was just all kinds of nasty (and broken, and left errors in the output).

I think @jbalboni's solution is the way to go. Because this didn't have the reference to `data['view:veteranId']`, I was assuming it was flattened...though in retrospect that wasn't a very astute assumption. 0.o That certainly makes life a lot less complicated!